users/engler/fairseq/training.py

```diff
@@ -166,6 +181,27 @@ def create_files(self):
         util.create_executable("fairseq.sh", self._get_run_cmd())
 
     def run(self):
+        if self.cached:
+            manifest_path = self.fairseq_hydra_config.config_dict["task"]["data"].get_path()
+            for name in ["train.tsv", "valid.tsv"]:
+                with open(f"{manifest_path}/{name}", "r") as manifest_file:
+                    manifest_lines = manifest_file.read().splitlines()
+                audio_path = manifest_lines[0]
+                bundle_lines = map(lambda line: audio_path + "/" + line.split("\t")[0], manifest_lines[1:])
+                with open(f"{name}.bundle", 'w') as bundle_file:
+                    bundle_file.write("\n".join(bundle_lines))
+                try:
+                    cached_audio_fn = sp.check_output(["cf", f"{name}.bundle"]).strip().decode("utf8")
+                except sp.CalledProcessError:
+                    print(f"Cache manager: Error occurred for {name}, using local file")
```

Comment:
Does this make sense? If the caching fails, you'd still have the path of the cached file in the new manifest, so you'd actually not use the local file, right?

Reply:
The message should also be adjusted because you're not using any local file. Printing some additional info before raising the error makes sense though.